test/cloud-canary/mzcompose.py

```diff
@@ -66,55 +69,105 @@ def workflow_default(c: Composition) -> None:
+        wait_for_cloud(c)
 
-        print("Obtaining mz_version() string from the cloud ...")
-        cloud_cursor = pg8000.connect(
-            host=cloud_hostname,
-            user=USERNAME,
-            password=APP_PASSWORD,
-            port=6875,
-            ssl_context=ssl.SSLContext(),
-        ).cursor()
-        cloud_cursor.execute("SELECT mz_version()")
-        cloud_version = cloud_cursor.fetchone()[0]
-        assert (
-            local_version == cloud_version
-        ), f"local version: {local_version} is not identical to cloud version: {cloud_version}"
-
-        print("Running smoke tests ...")
-        with c.override(
-            Testdrive(
-                default_timeout="600s",
-                materialize_url=materialize_url,
-                no_reset=True,  # Required so that admin port 6877 is not used
-            )
-        ):
-            c.run("testdrive", "*.td", rm=True)
+        version_check(c)
 
+        print("Running tests ...")
+        td(c, "*.td")
         test_failed = False
     finally:
         # Clean up
-        print(f"Shutting down region {REGION} ...")
-        c.run("mz", "region", "disable", REGION)
+        workflow_disable_region(c)
 
     assert not test_failed
+
+
+def cloud_hostname(c: Composition) -> str:
+    print("Obtaining hostname of cloud instance ...")
+    region_status = c.run("mz", "region", "status", REGION, capture=True)
+    sql_line = region_status.stdout.split("\n")[2]
+    cloud_url = sql_line.split("\t")[1].strip()
+    cloud_hostname = urllib.parse.urlparse(cloud_url).hostname
+    return cloud_hostname
+
+
+def wait_for_cloud(c: Composition) -> None:
+    print(f"Waiting for cloud cluster to come up with username {USERNAME} ...")
+    _wait_for_pg(
+        host=cloud_hostname(c),
+        user=USERNAME,
+        password=APP_PASSWORD,
+        port=6875,
+        query="SELECT 1",
+        expected=[[1]],
+        timeout_secs=600,
+        dbname="materialize",
+        ssl_context=ssl.SSLContext(),
+        # print_result=True
+    )
+
+
+def version_check(c: Composition) -> None:
+    print("Obtaining mz_version() string from local instance ...")
+    c.up("materialized")
+    local_version = c.sql_query("SELECT mz_version();")[0][0]
+
+    print("Obtaining mz_version() string from the cloud ...")
+    cloud_cursor = pg8000.connect(
+        host=cloud_hostname(c),
+        user=USERNAME,
+        password=APP_PASSWORD,
+        port=6875,
+        ssl_context=ssl.SSLContext(),
+    ).cursor()
+    cloud_cursor.execute("SELECT mz_version()")
+    cloud_version = cloud_cursor.fetchone()[0]
+    assert (
+        local_version == cloud_version
+    ), f"local version: {local_version} is not identical to cloud version: {cloud_version}"
+
+
+def td(c: Composition, *args: str) -> None:
+    materialize_url = f"postgres://{urllib.parse.quote(USERNAME)}:{urllib.parse.quote(APP_PASSWORD)}@{cloud_hostname(c)}:6875"
```

Comment:
```suggestion
    materialize_url = f"postgres://{urllib.parse.quote(USERNAME)}:{urllib.parse.quote(APP_PASSWORD)}@{urllib.parse.quote(cloud_hostname(c))}:6875"
```
Spaces will break, see for example:
```
$ psql postgres://materialize@local host:6875
psql: error: could not translate host name "local" to address: Temporary failure in name resolution
```
With my proposed change:
```
$ psql postgres://materialize@local%20host:6875
psql: error: could not translate host name "local host" to address: Name or service not known
```